### PR TITLE
Add Unit System awareness to WELSPECS reports

### PR DESCRIPTION
**Pull request by wito**
This feature adds Unit System awareness to WELSPECS reports, along with refactors from #1732.

### CODE REVIEW

File: opm/parser/eclipse/EclipseState/Schedule/Group/Group.hpp [hunk 1]
```diff
@@ -246,6 +246,7 @@ struct ProductionControls {
     const GroupProductionProperties& productionProperties() const;
     const std::map<Phase , GroupInjectionProperties>& injectionProperties() const;
     const GroupType& getGroupType() const;
+    const UnitSystem& getUnitSystem() const;
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Schedule/Group/Group.hpp:249]: That the `Group` object has access to a  `UnitSystem`  is more of an implementation accident - we should not cement this by using `Group` class a source of a `UnitSystem`. My suggested alternative is to pass it in as an argument to the `write_WELSPECS` function. At the callsite in `EclipseIO` the unit system can be had from:
```
const auto& unit_system = this->impl->es.getUnits();
```

**Comment by joakim-hove** [comment]
Apart from the comment about `Group` and `UnitSystem` this looks good and ready to be merged.

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
The three remaining `unimplemented<T>` fields can be obtained from this PR: https://github.com/OPM/opm-common/pull/1759 - with that PR in place the three columns can be obtained from:
```
const auto& perf_range = connection.perf_range();
double col1 = perf_range->first;
double col2 = perf_range->second;
double col3 = 0.50 * (col1 + col2);
```

**Comment by wito** [comment]
jenkins build this please

**Comment by wito** [comment]
jenkins build this please

**Comment by wito** [comment]
jenkins build this please

**Pull request merged**